lib/ops/opsservice/agents.go

```diff
@@ -298,13 +295,12 @@ func (r *AgentService) Wait(ctx context.Context, key ops.SiteOperationKey, numAg
 
 // AbortAgents shuts down remote agents and cleans up state
 func (r *AgentService) AbortAgents(ctx context.Context, key ops.SiteOperationKey) error {
-	group, err := r.peerStore.getGroup(key)
+	group, err := r.peerStore.removeGroup(ctx, key)
 	if err != nil {
 		return trace.Wrap(err)
 	}
-	err = group.Abort(ctx)
-	r.peerStore.removeGroup(ctx, key)
-	return trace.Wrap(err)
+	defer group.Close(ctx)
+	return trace.Wrap(group.Abort(ctx))
```

Comment:
Is the order change on purpose here? Before it was `group.Abort()` followed by `removeGroup()` and not it's the opposite.

Reply:
Yes, I changed order on purpose. I noticed that there's no need for a separate `getGroup` as all queries are done in the context of removing the group eventually as part of the operation so I combined these into a single removeGroup that does that - removes the group from the internal state but returns a reference to the value so it can be acted upon. There should be no change in semantics w.r.t functionality